docs/api-spec.md

```diff
@@ -48,269 +72,447 @@ There is no formal specification of the Kubernetes API and Resource Model. This
+Declares a parameter whose value has to be provided at runtime
 
-#### Spec
+| Field Name    | Field Type                  | Requirement         | Notes                                                                                                                                                                                    |
+| ------------- | --------------------------- | ------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `name`        | string                      | REQUIRED            |                                                                                                                                                                                          |
+| `description` | string                      | REQUIRED            |                                                                                                                                                                                          |
+| `type`        | [`ParamType`](#paramtype)   | REQUIRED (see note) | The values `string` and `array` for this field are REQUIRED, and the value `object` is RECOMMENDED.                                                                                      |
+| `properties`  | map<string,PropertySpec>    | RECOMMENDED         | `PropertySpec` is a type that defines the spec of an individual key. See how to define the `properties` section in the [example](../examples/v1/taskruns/beta/object-param-result.yaml). |
+| `default`     | [`ParamValue`](#paramvalue) | REQUIRED            |                                                                                                                                                                                          |
 
-| Field Name            | Field Type           | Requirement |
-|-----------------------|----------------------|-------------|
-| `params`              | `[]Param`            | REQUIRED    |
-| `taskSpec`            | `TaskSpec`           | REQUIRED    |
-| `workspaces`          | `[]WorkspaceBinding` | REQUIRED    |
-| `status`              | Enum:<br>- `""` (default)<br>- `"TaskRunCancelled"` | RECOMMENDED |
-| `timeout`             | string (duration)    | RECOMMENDED |
-| `serviceAccountName`^ | string               | RECOMMENDED |
+### ParamType
 
-^ In the Kubernetes implementation, `serviceAccountName` refers to a Kubernetes `ServiceAccount` resource that is assumed to exist in the same namespace. Other implementations MAY interpret this string differently, and impose other requirements on specified values.
+Defines the type of a parameter
 
-#### Status
+string enum, allowed values are `string`, `array`, and `object`. Supporting `string` is required while the other types are optional for conformance.
```

Comment:
This seems inconsistent with the `ParamSpec` section, where array is required

Reply:
good catch, fixed